src/components/AddressSearch.js

```diff
@@ -36,51 +35,44 @@ const defaultProps = {
 };
 
 const AddressSearch = (props) => {
-    const googlePlacesRef = useRef();
-    useEffect(() => {
-        if (!googlePlacesRef.current) {
-            return;
-        }
-
-        googlePlacesRef.current.setAddressText(props.value);
-    }, []);
+    const [skippedFirstOnChangeText, setSkippedFirstOnChangeText] = useState(false);
 
     const saveLocationDetails = (details) => {
         const addressComponents = details.address_components;
-        if (isAddressValidForVBA(addressComponents)) {
+        if (addressComponents) {
             // Gather the values from the Google details
-            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name');
-            const streetName = getAddressComponent(addressComponents, 'route', 'long_name');
+            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name') || '';
+            const streetName = getAddressComponent(addressComponents, 'route', 'long_name') || '';
+
+            const streetNumberAndName = `${streetNumber} ${streetName}`.trim();
+
             let city = getAddressComponent(addressComponents, 'locality', 'long_name');
             if (!city) {
                 city = getAddressComponent(addressComponents, 'sublocality', 'long_name');
                 Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
             }
-            const state = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
             const zipCode = getAddressComponent(addressComponents, 'postal_code', 'long_name');
+            const state = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
 
             // Trigger text change events for each of the individual fields being saved on the server
-            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
-            props.onChangeText('addressCity', city);
-            props.onChangeText('addressState', state);
-            props.onChangeText('addressZipCode', zipCode);
-        } else {
-            // Clear the values associated to the address, so our validations catch the problem
-            Log.hmmm('[AddressSearch] Search result failed validation: ', {
-                address: details.formatted_address,
-                address_components: addressComponents,
-                place_id: details.place_id,
-            });
-            props.onChangeText('addressStreet', null);
-            props.onChangeText('addressCity', null);
-            props.onChangeText('addressState', null);
-            props.onChangeText('addressZipCode', null);
+            if (streetNumberAndName.length > props.value.length) {
+                // We only replace what the user typed in the address input if we think ours is more complete
```

Comment:
Is "length" the marker of "completeness"? I'm having trouble following why we're looking at the lengths.

Reply:
It is because of this: https://expensify.slack.com/archives/C03TQ48KC/p1635967043023300?thread_ts=1635812089.454700&cid=C03TQ48KC

If the user types "123 Some Street #3232" (including the apt number), he will probably get a result like "123 Some Street". If the user clicks the result, we want try to avoid overwriting something that was more complete.

Reply:
Ok got it. I think the "length" was throwing me a bit since you could have a bunch of whitespace in this field making it look "more detailed" but that's probably uncommon and your explanation makes sense.